AutoPartsShop.API/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,140 @@
+﻿using AutoPartsShop.API.Extensions;
+using AutoPartsShop.API.Repositories.Interfaces;
+using AutoPartsShop.Models.DTOs;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AutoPartsShop.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductsController : ControllerBase
+    {
+        private readonly IProductRepository productRepository;
+
+
+        public ProductsController(IProductRepository productRepository)
+        {
+            this.productRepository = productRepository;
+        }
+
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<ProductDTO>>> GetItems()
+        {
+            try
+            {
+                var products = await this.productRepository.GetItems();
```

Comment:
The comments I had on the previous PR still apply. Best practices say that we should split the code into multiple classes/layers which have different functionalities.

Here, it should be something like:
- AutoPartsShop.API which contains the Controllers
- AutoPartsShop.Application which contains Services (where the logic from here should be moved)
- AutoPartsShop.DataAccess which contains Repositories (which have as a single responsibility to connect to the database)